src/Language/PureScript/Sugar/Names.hs

```diff
@@ -269,17 +269,29 @@ renameInModule imports (Module modSS coms mn decls exps) =
       throwError . errorMessage' pos $ OverlappingNamesInLet
     return ((pos, args ++ bound), Let w ds val')
   updateValue (_, bound) (Var ss name'@(Qualified Nothing ident)) | ident `notElem` bound =
-    (,) (ss, bound) <$> (Var ss <$> updateValueName name' ss)
+    updateOrReplaceValue (ss, bound) name' updateValueName' (Var ss) showIdent
   updateValue (_, bound) (Var ss name'@(Qualified (Just _) _)) =
-    (,) (ss, bound) <$> (Var ss <$> updateValueName name' ss)
+    updateOrReplaceValue (ss, bound) name' updateValueName' (Var ss) showIdent
   updateValue (_, bound) (Op ss op) =
-    (,) (ss, bound) <$> (Op ss <$> updateValueOpName op ss)
+    updateOrReplaceValue (ss, bound) op updateValueOpName' (Op ss) showOp
   updateValue (_, bound) (Constructor ss name) =
-    (,) (ss, bound) <$> (Constructor ss <$> updateDataConstructorName name ss)
+    updateOrReplaceValue (ss, bound) name updateDataConstructorName' (Constructor ss) runProperName
   updateValue s (TypedValue check val ty) =
     (,) s <$> (TypedValue check val <$> updateTypesEverywhere ty)
   updateValue s v = return (s, v)
 
+  updateOrReplaceValue
+    :: (SourceSpan, [Ident])
+    -> Qualified a
+    -> (Qualified a -> SourceSpan -> m (Maybe (Qualified a)))
+    -> (Qualified a -> Expr)
+    -> (a -> Text)
+    -> m ((SourceSpan, [Ident]), Expr)
+  updateOrReplaceValue (ss, bounds) name updateExpr mkExpr showName = do
```

Comment:
Do you think it would make more sense to call the third argument `updateName`?

Reply:
Yep, I think it does